### PR TITLE
[gateway] Move approve to gateway @open sesame 04/22  13:37

### PR DESCRIPTION
**Pull request by zhoonit**
- [gateway] Move approve to gateway 

```
This patch moves approve comment to gateway since approve should wait for
prebuild group to be successful as well

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

**Comment by zhoonit** [comment]
verified with PR https://github.com/nnstreamer/nntrainer/pull/1119

File: ci/taos/checker-pr-gateway.sh [hunk 1]
```diff
@@ -210,18 +210,37 @@ function run_all_checkers(){
...
+    echo -e "[DEBUG] waiting... pid: ${pid_}"                         | tee -a $logfile_gateway
+    wait $pid_
+    status+=$?
+  done
+
+  # this means every checker was successful
+  if [[ $status == 0 ]]; then
+    # Let's approve it as a reviewer if this PR passes all CI modules.
+    if [[ $pr_comment_review_activity == 1 ]]; then
+        message=" $user_id, :100: All CI checkers are successfully verified. Thanks."
```
Comment (gichan-jang) [inline ci/taos/checker-pr-gateway.sh:240]: `user_id` not defined here.
Let's add
```
# Get user ID from the input_repo string
set -- "${input_repo}"
IFS="\/"; declare -a Array=($*); unset IFS;
user_id="@${Array[3]}"
```

File: ci/taos/checker-pr-gateway.sh [hunk 2]
```diff
@@ -210,18 +210,37 @@ function run_all_checkers(){
     # Note that you must modify the Out-of-PR (OOP) killer because OOP killer depends on synchronous method.
     pushd ./taos/
     ${cmd_list[$i]} $checker_args                                        | tee -a $logfile
+    status+=${PIPESTATUS[0]}
```
Comment (kparichay) [inline ci/taos/checker-pr-gateway.sh:213]: move this to below line 215?
now line 215 gets status of this line.

Comment (zhoonit) [reply ci/taos/checker-pr-gateway.sh:213]: 215 gets the recent background pid, unless we do `${cmd_list[$i]} $checker_args | tee -a $logfile &`, it remains 0.

if it is on the background, line 213 operation is not background so it does no affect line 215

File: ci/taos/checker-pr-gateway.sh (hunk 1 above)
Comment (zhoonit) [reply ci/taos/checker-pr-gateway.sh:245]: I see, thank you :)

**Review by gichan-jang — APPROVED** [review]
LGTM

**Review by leemgs — APPROVED** [review]
I like this PR to handle pipelined processes correctly. If anyone does not have additional comments, I am going to merge this PR finally. 😄

**Pull request merged**